ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientBase.java

```diff
@@ -45,16 +46,19 @@ protected String generateEndpointSuggestion(String existingSuggestedEndpointUri,
             URIBuilder existingEndpoint;
             try {
                 existingEndpoint = new URIBuilder(dataSource);
-                endpointUriToSuggestStr = emendEndpointUri(existingEndpoint);
+                endpointUriToSuggestStr = amendEndpointUri(existingEndpoint);
             } catch (URISyntaxException e) {
                 log.error("Couldn't parse dataSource '{}', so no suggestion can be made.", dataSource, e);
             }
+            catch (IllegalArgumentException e) {
+                log.error("URI is already in the correct format '{}', so no suggestion can be made.", dataSource, e);

```

Comment:
It's specifically a URL. A URI is much broader.

I'm debating whether this is worth erroring, or warning. WDYT?

Reply:
I think it should be an error. 99% the user has made a mistake, I want them to be able to find it easily